website/docs/r/launch_template.html.markdown

```diff
@@ -112,7 +112,7 @@ The following arguments are supported:
   `vpc_security_group_ids` instead.
 * `vpc_security_group_ids` - A list of security group IDs to associate with.
 * `tag_specifications` - The tags to apply to the resources during launch. See [Tags](#tags) below for more details.
-* `user_data` - The user data to provide when launching the instance.
+* `user_data` - The Base64-encoded user data to provide when launching the instance.
```

Comment:
Does EC2 Base64 decode this automatically?

Reply:
Yep. It even shows plain text in AWS console

Reply:
Interesting -- we've needed to implement this as a separate attribute (`user_data_base64`) for it to work in other places. Usually Terraform shows a difference due to the API returning it differently depending on whether its Base64 or not.

Reply:
An easy way to verify this is with https://www.terraform.io/docs/providers/template/d/cloudinit_config.html with `gzip` and `base64_encode` enabled.

Reply:
Well, from docs https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestLaunchTemplateData.html launch template accepts only base64 . So you will need to use base64encode interpolation when providing it. In can probably add this to examples

Reply:
Great find! 👍